RobloxIPFinder/Form1.cs

```diff
@@ -19,34 +21,54 @@ public Form1()
         }
 
         // Made by daymon - twitter.com/disband
+
+        // Fixed by Nick.
+        
         private void simpleButton1_Click(object sender, EventArgs e)
         {
-            int counter = 0;
-            string line;
-            string appDataFolder = Environment.GetEnvironmentVariable("LocalAppData");
-            string filePath = Path.Combine(appDataFolder, "Roblox\\logs");
-            DirectoryInfo info = new DirectoryInfo(filePath);
-            FileInfo[] files = info.GetFiles().OrderByDescending(p => p.CreationTime).ToArray();
-            foreach (FileInfo fil in files)
-            {
-                if(Convert.ToString(fil).Contains("GameStartScript"))
-                {
-                    filePath = Path.Combine(filePath, Convert.ToString(fil));
-                    break;
-                }   
-            }
-            System.IO.StreamReader file = new System.IO.StreamReader(filePath);
-            while ((line = file.ReadLine()) != null)
+
+            string LocalAppDataPath = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+
+            string FilePath = Path.Combine(LocalAppDataPath, "Roblox", "logs");
+
+            DirectoryInfo Info = new DirectoryInfo(FilePath);
+            FileInfo[] Files = Info.GetFiles().OrderByDescending(f => f.CreationTime).ToArray();
+
+            string Line;
+
+            var files = new DirectoryInfo(FilePath).GetFiles("*.*");
+            string latestfile = " ";
+            DateTime lastupdated = DateTime.MinValue;
+            int count = 0;
+            foreach (FileInfo file in files.Reverse())
             {
-                if(line.Contains("Connecting to"))
+                if (file.LastWriteTime > lastupdated)
                 {
-                    int index = line.IndexOf("to") + 2;
-                    string piece = line.Substring(index);
-                    MessageBox.Show(piece);
+                    lastupdated = file.LastWriteTime;
+                    count++;
+                    Console.WriteLine(" LatestFileName : " + file.Name + " " + count);
+                    //Console.ReadLine();
+                    {
+                        latestfile = file.Name;
+
+                        StreamReader FileZ = new StreamReader(new FileStream(file.FullName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite));
+
+                        while ((Line = FileZ.ReadLine()) != null)
+                        {
+                            Console.WriteLine(Line);
+
+                            if (Line.Contains("Connecting to"))

```

Comment:
Instead of checking for `Connecting to`, you could go down a bit more and look for the `serverId:`. It seems that `Connecting to` could be wrong if the user fails to connect- wheres `serverId` is only logged *after* the user connects. This regex would also work, but whether to use regex or not is up to design choices:
```c#
new Regex("(?<=serverId: )(?<IPAddress>\\d{1,3}.\\d{1,3}.\\d{1,3}.\\d{1,3})\\|(?<Port>\\d{1,5})");
```

Reply:
Ah good to know. Thanks